### PR TITLE
Styles change and bug fix branch 2

### PR DESCRIPTION
**Pull request by SonaliBedge**
I made small changes to a few files and added a rating feature for the courses.

Added a course type option for users to search for courses based on their course type preferences. Added pagination option too.

image

Ratings for all courses are displayed based on user feedback. Only courses with lessons have ratings available.

image

Added a back button to the form for users to return to the previous page, and updated the style and font. 

image

same here - Added a back button to the form for users to return to the previous page, and updated the style and font.

image

### CODE REVIEW

File: src/components/layout/Nav.jsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import SignUp from './SignUp';
 import styles from './Nav.module.css';
 import { useAuth } from '../../AuthContext';
 import logo from '../../assets/logos/blue.png';
+// import Profile from '../../pages/UserDashboard/Profile';
```
Comment (lauradurandv) [inline src/components/layout/Nav.jsx:8]: a the console logs

File: src/pages/CourseDetails/CourseDetails.module.css [hunk 2]
```diff
@@ -98,6 +101,32 @@
   animation: spin 1s linear infinite;
   margin: 0 auto;
 }
+.backButton {
```
Comment (lauradurandv) [inline src/pages/CourseDetails/CourseDetails.module.css:104]: be careful with the px's

File: src/pages/Learn/Lesson.module.css [hunk 3]
```diff
@@ -51,13 +51,30 @@
...
+  color: white;  
 }
 
 .descriptionTitle:hover {
-  background-color: #debf41;
-  transform: scale(1.02);
+  background-color: #debf41; 
+}
+.highlightedWord {
+  background-color: yellow; /* Highlight the currently spoken word */
```
Comment (lauradurandv) [inline src/pages/Learn/Lesson.module.css:61]: cool!

File: src/pages/UserDashboard/Courses.module.css [hunk 4]
```diff
@@ -88,3 +83,125 @@
...
+.categoryCard:hover {
+  background-color: #8e861a;
+}
+
+.activeCategory {
+  border-color: #007bff;
+  background-color: #0f437b;
+  color: white;
+}
+.errorMessage {
```
Comment (lauradurandv) [inline src/pages/UserDashboard/Courses.module.css:154]: px's here too

File: src/components/layout/Nav.jsx (hunk 1 above)
Comment (SonaliBedge) [reply src/components/layout/Nav.jsx:8]: Where are the console logs, Laura? Do you mean to remove the commented code?

File: src/pages/CourseDetails/CourseDetails.module.css (hunk 2 above)
Comment (SonaliBedge) [reply src/pages/CourseDetails/CourseDetails.module.css:104]: I will convert into 'rem'

File: src/pages/Learn/Lesson.module.css (hunk 3 above)
Comment (SonaliBedge) [reply src/pages/Learn/Lesson.module.css:61]: Thank you!

File: src/pages/UserDashboard/Courses.module.css (hunk 4 above)
Comment (SonaliBedge) [reply src/pages/UserDashboard/Courses.module.css:154]: I will change to 'rem'

**Review by EdithHarrison — APPROVED** [review]

**Comment by EdithHarrison** [comment]
Can we make the doubleheader into a sidebar instead?? to make it consistent with the teacher's dashboard?

**Pull request merged**